extensions/objects/templates/numeric-expression-editor.component.ts

```diff
@@ -41,23 +41,24 @@ angular.module('oppia').component('numericExpressionEditor', {
       ctrl.hasBeenTouched = false;
 
       ctrl.isCurrentAnswerValid = function() {
-        if (ctrl.hasBeenTouched) {
-          // Replacing abs symbol, '|x|', with text, 'abs(x)' since the symbol
-          // is not compatible with nerdamer or with the backend validations.
-          ctrl.value = MathInteractionsService.replaceAbsSymbolWithText(
+        // Replacing 'absolutevalue(x)' with 'abs(x)', 'neg(x)' with '-(x)'
+        // and 'squareroot(x)' with 'sqrt(x)' for compatibility with nerdamer
+        // and backend validators.
+        ctrl.value = ctrl.value.replace(/absolutevalue\(/g, 'abs(');
+        ctrl.value = ctrl.value.replace(/neg\(/g, '-(');
```

Comment:
Why is neg handled here but not in the init() case above?

This is why we need a constant -- although now I'm thinking it actually ought to be a service, since you've got stuff after this that is common between all the invocations, too.

Reply:
This code is no longer needed and I've removed it from all three interactions.